http-client/src/main/java/com/tutrit/httpclient/HttpEngineerGateway.java

```diff
@@ -0,0 +1,147 @@
+package com.tutrit.httpclient;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.tutrit.bean.Engineer;
+import com.tutrit.config.ConfigProvider;
+import com.tutrit.gateway.EngineerGateway;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.util.Optional;
+
+@Component
+public class HttpEngineerGateway implements EngineerGateway {
+    private static final String CONTENT_TYPE = "application/json";
+    private static final String PATH = "engineers";
+    private final ObjectMapper objectMapper;
+    private final ConfigProvider config;
+    private final HttpClient httpClient;
+
+    public HttpEngineerGateway(
+            final ConfigProvider config,
+            final HttpClient httpClient,
+            final ObjectMapper objectMapper
+    ) {
+        this.objectMapper = objectMapper;
+        this.config = config;
+        this.httpClient = httpClient;
+    }
+    @Override
+    public Engineer saveEngineer(Engineer engineer) {
+        validateEngineer(engineer);
+
+        String url = "%s/%s/%s".formatted(config.getUrl(), PATH, engineer.engineerId());
+        HttpRequest request = HttpRequest.newBuilder()
+                .uri(URI.create(url))
+                .header("Content-Type", CONTENT_TYPE)
+                .POST(createEngineerBodyPublisher(engineer))
+                .build();
+
+        sendHttpRequest(request);
+        return engineer;
+    }
+
+    private void validateEngineer(Engineer engineer) {
+        if (engineer == null) {
+            throw new IllegalArgumentException("Engineer can't be null");
+        }
+
```

Comment:
validation should be moved to separate class. 
you can create validator functional interface, and then autowire all validators in Collestion in service, and then just go one by one. 

So validation would be a separate story, please create jira ticket

Reply:
done